### ISSUE TITLE
Be able to put Mesh Vertex and Index Buffers into a ComputeBuffer?

### ISSUE THREAD

**Issue by MD485** [Facepunch/sbox-issues#4551]
### For?

S&Box, Other

### What can't you do?

I wanna make dynamic grass that sways in the wind.  
More specifically to generate and manipulate a large mesh with ComputeShaders.  

The issue is downloading it from the GPU to the CPU with `ComputeBuffer.GetData`, packaging it into a mesh, then uploading that mesh back to the GPU every frame through a model renderer, causes such a performance hit it makes generating the meshes GPU side somewhat pointless, and performance wise it would be impractical to do on the CPU.

### How would you like it to work?

So Unity has this functionality:

https://docs.unity3d.com/ScriptReference/Mesh.GetIndexBuffer.html
https://docs.unity3d.com/ScriptReference/Mesh.GetVertexBuffer.html
```cs
using UnityEngine;

public class ExampleScript : MonoBehaviour
{
    public Mesh mesh;
    public ComputeShader computeShader;
    void Start()
    {
        // Mark the buffers as needing "Raw" (ByteAddressBuffer, RWByteAddressBuffer in HLSL shaders) access.
        mesh.indexBufferTarget |= GraphicsBuffer.Target.Raw;
        mesh.vertexBufferTarget |= GraphicsBuffer.Target.Raw;
        // Get the buffers of the Mesh, and set it up as a buffer parameter to a compute shader.
        var indexBuffer = mesh.GetIndexBuffer();
        var vertexBuffer = mesh.GetVertexBuffer(0);
        computeShader.SetBuffer(0, "MeshIndexBuffer", indexBuffer);
        computeShader.SetBuffer(0, "MeshVertexBuffer", vertexBuffer);
        indexBuffer.Dispose();
        vertexBuffer.Dispose();
    }
}
```
They seem to be using a ByteAddressBuffer as the backing for the two buffers when the flag is set.

With the flexibility to be able to alter meshes with a compute shader, you can do some quite cool stuff:
https://github.com/keijiro/ComputeMarchingCubes

Plus it would make DrawMeshInstancedIndirect() from the unity docs for really easy for anyoneto write.
https://docs.unity3d.com/ScriptReference/Graphics.DrawMeshInstancedIndirect.html
Which is how youtube tutorials do things like render millions of cubes at once.
https://www.youtube.com/watch?v=vhCPb2SzakU

### What have you tried?

Well I tried this:
```cs
var vertsPointer = CollectionsMarshal.AsSpan( verts );
vertBuffer = new ComputeBuffer<Vertex>( verts.Count, ComputeBufferType.ByteAddress );
mesh.SetVertexBufferData( vertsPointer, 0 );
vertBuffer.SetData( vertsPointer, 0 );
shaderImpl.Attributes.Set( "Verts", vertBuffer );
```
But sbox started showering me with error messages, so I imagine I wasn't supposed to do that.

I can't really imagine any way to do what I'm trying to do other than having my Index and Vertex Buffer as ComputeBuffers.

### Additional context

https://docs.google.com/document/d/1_YrJafo9_ZsFm4-8K2QlD0k3RgwZ_49tSA84paobfcY/edit
Unity released this short google doc about their new Mesh API back when they released it for added context, but there's not much there really.

**Comment by handsomematt** [comment]
Yeah this all makes sense and is simple enough for us to expose for you to do - I'm going to be adding it when I tackle terrain detailing.
